src/auth/oidc-react.ts

```diff
@@ -0,0 +1,386 @@
+  return user as unknown as ClientUser;
+}
+
+function bindEvents(
+  manager: UserManager,
+  eventFunctions: {
+    onAuthChange: Client['onAuthChange'];
+    setError: ClientFactory['setError'];
+    eventTrigger: ClientFactory['eventTrigger'];
+  }
+): void {
+  const { onAuthChange, setError, eventTrigger } = eventFunctions;
+  manager.events.addUserLoaded((): void =>
+    eventTrigger(ClientEvent.CLIENT_AUTH_SUCCESS)
+  );
+  manager.events.addUserUnloaded((): boolean => onAuthChange(false));
+  manager.events.addUserSignedOut((): boolean => onAuthChange(false));
+  manager.events.addUserSessionChanged((): boolean => onAuthChange(false));
+  manager.events.addSilentRenewError((renewError?: Error): void => {
+    const errorObj = renewError || undefined;
+    const message = errorObj ? errorObj.message : '';
+    setError({
+      type: ClientError.AUTH_REFRESH_ERROR,
+      message,
+    });
+  });
+  manager.events.addAccessTokenExpired((): void =>
+    eventTrigger(ClientEvent.TOKEN_EXPIRED)
+  );
+  manager.events.addAccessTokenExpiring((): void =>
+    eventTrigger(ClientEvent.TOKEN_EXPIRING)
+  );
+}
+
+export function getSessionStorageKey(clientConfig?: ClientConfig): string {
+  const config = clientConfig || getClientConfig();
+  return `oidc.user:${config.authority}:${config.clientId}`;
+}
+
+export function createOidcClient(): Client {
+  if (!hasValidClientConfig()) {
+    const errorMessage = 'Invalid client config';
+    // eslint-disable-next-line no-console
+    console.error(errorMessage, getClientConfig());
+    throw new Error(errorMessage);
+  }
+  const clientConfig = getClientConfig();
+  const oidcConfig: UserManagerSettings = {
+    userStore: new WebStorageStateStore({ store: window.sessionStorage }),
+    authority: clientConfig.authority,
+    automaticSilentRenew: clientConfig.automaticSilentRenew,
+    client_id: clientConfig.clientId,
+    redirect_uri: getLocationBasedUri(clientConfig.callbackPath),
+    response_type: clientConfig.responseType,
+    scope: clientConfig.scope,
+    silent_redirect_uri: getLocationBasedUri(clientConfig.silentAuthPath),
+    post_logout_redirect_uri: getLocationBasedUri(clientConfig.logoutPath),
+  };
+  const manager = new UserManager(oidcConfig);
+  const {
+    eventTrigger,
+    getStoredUser,
+    setStoredUser,
+    fetchApiToken,
+    ...clientFunctions
+  } = createClient();
+
+  const { isAuthenticated, isInitialized, setStatus, getStatus, setError } =
+    clientFunctions;
+  if (clientConfig.enableLogging) {
+    Oidc.Log.logger = console;
+    Oidc.Log.level = Oidc.Log.DEBUG;
+  }
+
+  const getSessionStorageData = (): AnyObject | undefined => {
+    const userKey = getSessionStorageKey(clientConfig);
+    const storedString = sessionStorage.getItem(userKey);
+    if (
+      !storedString ||
+      storedString.length < 2 ||
+      storedString.charAt(0) !== '{'
+    ) {
+      return undefined;
+    }
+    try {
+      return JSON.parse(storedString);
```

Comment:
try {
      const parsedData = JSON.parse(storedString);
     return parsedData instance of Array ? undefined  : parsedData
} catch (e) {
      return undefined;
    }
  };

Reply:
Now as I look into it, I think the original solution is actually a simple and effective way to check the stored data. It's not only array that we need to rule out, but also null, boolean, string, etc. So we'll need to do something like:

```
const parsedData = JSON.parse(storedString);
if (parsedData === null || Array.isArray(parsedData) || typeof parsedData === 'boolean' || typeof parsedData === 'string'  || typeof parsedData === 'number') {
    return undefined.
}
```